app/Http/Controllers/AuthController.php

```diff
@@ -2,34 +2,93 @@
 
 namespace App\Http\Controllers;
 
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Validator;
+use App\Components\ForgotPasswordHandler;
+use App\Components\Services\PasswordResetService;
+use App\Components\Errors\UnauthorizedException;
 use App\Components\AuthHandler;
 use App\Components\Services\UserService;
 use App\Components\TokenGenerator;
+use App\Components\Traits\HttpResponse;
+use App\PasswordReset;
 use App\User;
-use Illuminate\Http\Request;
-use Illuminate\Support\Facades\Validator;
 
 class AuthController extends Controller
 {
+    use HttpResponse;
+
+    private $status = 500;
+
     public function login(Request $request)
     {
-        $validated = Validator::make($request->all(), [
+        $requestContent = Validator::make($request->all(), [
             'email' => 'required|email',
             'password' => 'required|min:6',
             'device_name' => 'required',
         ])->validate();
 
         try {
             $service = new UserService(new User());
-            $authHandler = new AuthHandler($service, new TokenGenerator());
+            $handler = new AuthHandler($service, new TokenGenerator());
             
-            $responseContent = $authHandler->authenticate($validated);
+            $responseContent = $handler->authenticate($requestContent);
 
             return response()->json($responseContent);
         } catch (\Exception $error) {
+            return $this->respondWithError($error);
+        }
+    }
+
+    public function sendResetPasswordRequest(Request $request)
+    {
+        $requestContent = Validator::make($request->all(), [
+            'email' => 'required|email'
+        ])->validate();
+
+        $user = User::where('email', $requestContent['email'])->first();
+        
+        try {
+            if (! $user) {
+                throw new UnauthorizedException();
+            }
```

Comment:
Isso aqui pra min parece um not-found ou ``` FirstOrFail```, pq o se o cara passou um email de um usuário que não existe então foi erro dele, ou seja, ele não ta desautorizado. O erro 401 é quando credências de acesso não permitem o usuário acessar determinado recurso.

Reply:
Certo, é que em outra PR surgiu o mesmo if, e uma request foi que deveria ser unauthorized, no login salve me engano. Acabei abstraindo pra aqui. Vou mudar!